src/index.rs

```diff
@@ -572,6 +648,36 @@ impl Index {
                 .is_some()
         })
     }
+
+    /// adds the given constant expression to the constants arena and returns the ID to reference it
+    pub fn add_constant_expression(&mut self, expr: AstStatement) -> ConstId {
+        self.constant_expressions.add_expression(expr)
+    }
+
+    /// convinience-method to add the constant exression if there is some, otherwhise not
+    pub fn add_maybe_constant_expression(&mut self, expr: Option<AstStatement>) -> Option<ConstId> {
```

Comment:
how do you feel about maybe_add.. (or try_add) try_xx is used in rust for conversions, but usually gives back a result

Reply:
renamed and added further documentation explaining that this is only for the convenient situation when you have an Option<AstStatement> - e.g. optional initial values -> we have a lot of such situations